File: TUF/Numbers/CheckHarshadNumber.java
import java.util.Scanner;

public class CheckHarshadNumber {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the number to check: ");
        int num = input.nextInt();
        input.close();
        System.out.println(checkIsHarsadNumber(num));
    }

    private static boolean checkIsHarsadNumber(int num) {
        int sum = 0;
        int dupNum = num;
        while (num != 0) {
            int rem = num % 10;
            sum += rem;
            num = num / 10;
        }
        return dupNum % sum == 0;
    }
}

File: TUF/Numbers/CheckHarshadNumber.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function checkIsHarsadNumber(num) {
    var sum = 0;
    var dupNum = num;
    while (num != 0) {
        var rem = num % 10;
        sum += rem;
        num = Math.floor(num / 10);
    }
    return dupNum % sum == 0;
}
var num = 378;
console.log("".concat(num, " is a harsad number: ").concat(checkIsHarsadNumber(num) ? "Yes" : "No"));
